src/DFG.cpp

```diff
@@ -175,6 +175,44 @@ void DFG::combine(string t_opt0, string t_opt1) {
   }
 }
 
+void DFG::combineIter(list<string>* cPattern, list<DFGNode*>* cPatternNodes) {
+  string t_opt = cPattern->front();
+  cPattern->pop_front();
+  DFGNode* dfgNode = cPatternNodes->back();
+  for (DFGNode* succNode: *(dfgNode->getSuccNodes())) {
+    if (succNode->isOpt(t_opt) and !succNode->hasCombined()) {
+      if (cPattern->empty() == true and succNode->isSuccessorOf(cPatternNodes->front())){
+        // succNode->isSuccessorOf(cPatternNodes->front()) is not needed if the pattern is not a cycle
+        cPatternNodes->push_back(succNode);
+        DFGNode* opt0Node = cPatternNodes->front();
+        opt0Node->setCombine();
+        cPatternNodes->pop_front();
+        for(DFGNode* optNode: *cPatternNodes){
+          opt0Node->addPatternPartner(optNode);
```

Comment:
We do need a way to revoke the wrong fusion/combination. We cannot just `setCombine()` whenever a node is visited. Say you want to fuse `phi-add`, but there is a cycle of `phi-add-sub`, can the wrong fusion be avoided.

Reply:
So our code only fuses a cycle, right? If there is a cycle of `phi-add-sub`, we won't fuse it into `phiadd-sub`. Mine already avoids this situation using `if (cPattern->empty() == true and succNode->isSuccessorOf(cPatternNodes->front()))`. This guarantees the head and tail of the target pattern are connected.

Reply:
Ok. Please leave comments on top of the function that explicitly states the pattern provided by the user should be a cycle.

Reply:
and "otherwise, the fusion won't be performed".